File: src/scp.lua
#!/usr/bin/env lua

local symbol
local env_symbol = os.getenv("CCC_SYMBOL")

if arg[1] then
    symbol = arg[1]
else
    if env_symbol ~= nil then
        symbol = env_symbol
    else
        symbol = ""
    end
end

local function color(col)
    return ("\027[38;5;%sm"):format(col)
end

io.write("\n")

for i = 0,7 do
    local color = (" \027[%sm%s \027[m"):format(30+i, symbol)
    io.write(color)
end

io.write("\n")


for i = 0,7 do
    local color = (" \027[%sm%s \027[m"):format(90+i, symbol)
    io.write(color)
end


io.write("\n\n")

File: src/scp.js
#!/usr/bin/env node
const process = require("node:process");

function main() {
  let symbol;
  if (process.argv.length >= 3) {
    symbol = process.argv[2];
  } else if (process.env.CCC_SYMBOL) {
    symbol = process.env.CCC_SYMBOL;
  } else {
    symbol = "\ue74e";
  }
  console.log();
  for (let i = -1; i < 7; i++) {
    process.stdout.write("\033[" + (31 + i) + "m" + symbol + "\033[m ");
  }
  console.log();
  for (let i = -1; i < 7; i++) {
    process.stdout.write("\033[" + (91 + i) + "m" + symbol + "\033[m ");
  }
  console.log();
}

main();
